### DISCUSSION TITLE
Request for comment: Redesign Darwinia Tokenomics in the philosophy of Ethereum EIP1559

### DISCUSSION THREAD

**Discussion by hackfisher** [darwinia-network/darwinia#1163] [General]
# Introduction
This is to propose a new design of the tokenomics for Darwinia's native token RING, to better fit the network's economic, incentive and utilitity purpose of the token and the competition in the free market of tokens.

The ideas and lessons was learned from Ethereum's improvement of tokenomics during switch from POW to POS, especially from EIP1559 (Ref.2).

# Problem and Solution
Current tokenomics is described  in Darwinia's genepaper V2 (Ref.1) as following:

> RING's initial supply (INITIAL_SUPPLY) is 2 billion, after the initial supply is generated, the block reward of year N is 1 - (99 /100)^sqrt(N) of total remaining issuable, until reaching the hard cap of the supply which is 10 billion.
> 
> Total remaining issuable RING = HARD_CAP - CURRENT_SUPPLY

<img width="457" alt="image" src="https://github.com/darwinia-network/darwinia/assets/1070122/938c86f6-d65a-4f6c-b4e6-57b161f33fb4">
<img width="461" alt="image" src="https://github.com/darwinia-network/darwinia/assets/1070122/99f193c0-d3e0-4f24-b4b8-93ea33a9d7a2">

The largest problem of current implementation/design is that it introduced a model of target supply, to be more specific, no matter how many gas or tokens are used and burned in the network, the current supply design always set the target supply to be one had cap, and the supply and inflation in specific year is set by that formula while they have no relationship with the performance and value capture, this will make the network activities very hard to be reflected to the token economic. To resolve this problem, the new design inspired by Ethereum EIP 1559will separate the token supply changes to two part, the burn part and new minted part, in one hand, all the gas and tokens fees captured by the network should be directly burned from the supply, in another hand, there will be a simple and fixed block reward which will be minted as new tokens to support the security and growth of the network, this block reward in every year or every day will be a fix amount to support following parts:
- The consensus security of the network. Current Darwinia2 is working as a parachain of Polkadot, then a ratio of the block reward will be allocated to treasury for future bidding of the parachain slot from Polkadot, and Polkadot will help guard the consensus security of Dawinia2, Polkadot may also provide some additional benefits, such as XCM cross-chain message service for parachains. There are still challenges to predict the cost of parachain slots from Polkadots, because it is very flexible and depend on the demands and supply, so for Darwinia2 as sovereign network, using the security budget from block reward to bidding consensus security using alternative solutions is possible and deserve research.
- Liveness safety and support. Current Darwinia2 is allocating 40% of the block reward to the staking of collators, this is to keep there are at least one honest collator is producing blocks for the networks, that is liveness.
- Ecosystem Growth Treasury. Bidding parachain slots might only be the partial use of the Treasury, there could be other proposal and grants for supporting the buidling of the network, the integration of the infrastructure, the development of the ecosystem, and the growth of the community etc.

There should be sufficient budget to support the security of the network regardless of the gas fees and value capture of the network as that is is a prerequisite to ensure the normal operation of the network against attacks, while the income of the network is greatly affected by the market and external influences as the market cannot be good or bad from time to time.

Another problem of the current design is that, we can see from the above curve chart, the inflation rate in beginning years is relative high and arbitrary, this will lead to a decline in attractiveness and competitiveness in the token market, which in turn will affect community and ecological development. So the redesigned model should be have less/comparable inflation rate and supply compared to the existing design, at least in the next 100 years.

Based on the above two problems and solution constrains, here we propose a new design for the Darwinia.

1. The gas and fees, or other income values captured by the network will be directly burned.
2. For the block reward supporting the network security and treasury, there will be a 80M RINGs each year, around 219K RING every day, specifically so far, will include the budget for bidding Polkadot parachain, staking for collators, treasury remaining.

Here is a summary of the maximum token supply and inflation rate over time for the given token supply design with an initial supply of 2.275 billion tokens and an annual minting of 80 million tokens, extended to a range of 100 years:

![New design's inflation rate](https://www.wolframcloud.com/obj/ccddea7b-5112-41a9-bb10-356d511a7e5d)
![New design's token supply](https://www.wolframcloud.com/obj/9a6a8bfa-9e6f-4bd6-bacd-3094e50817bc)

Actually, the maximum curve is the worst case with no RINGs being burned, tokens being in that year will reduce the supply and inflation rate, if there are more than 80M RING burned in that year, the inflation rate will be below zero, and become deflation, just like what happened with the Ethereum network, you can experiment with the parameters using [ultrasound.money](https://ultrasound.money/)(Ref.3) to learn how the network activities affect the token economic.

References:
1. https://darwinia.network/genepaper_v2.pdf
4. https://eips.ethereum.org/EIPS/eip-1559
5. https://ultrasound.money/

**Comment by SasoLithops** [comment]
Hi!

I think it is time for community to discuss Darwinia tokenomics adjustment. I had concerns that we are not taking different variables in equasion when we did 60:40 reward split. What is happening is as predicted we get double the rewards but token price keeps on dropping. I have to admit since that decision has been made I am selling my Ring rewards weekly into DOT as I fear token price will go to many zeros. I am being honest for the love of project,not to stab it in back. I still haven't sold any Ring or Kton prior to that decision. And will switch DOT back to KTON if I belive in improvement.
A proper calculations need to be made to improve tokenomics. To calculate all the variables that could effect the potential circulating supply. I understand we can not predict the price that is NOT my intention.My intention is to improve tokenomics to make it more attractive for long term hold.

**Reply by hackfisher** [reply]
I agree, to be more clear, the purpose of changing the tokenomics design is to promote the correct token-economic feedback loop, and to guide the correct expectations, just like any other monetary policy. Also, the policy can NOT change intrinsic value(except scarcity) let alone directly affect on price, there are more other factors taking effects at the same, such as outside economic, liquidity, speculation, product-market adoption etc.

Changing expectations and feedback loops starts with what can be done, tokenomics is one of them, better design can make incentives more effective and resource allocation more reasonable, indirectly affect the growth, cost and expectations of the network in the crypto market.

But before it goes into effect, the following conditions are expected to be in place:
1. Obtain a majority consensus to avoid being replaced by other consensus and pay too high a cost to the community, whether due to education or misunderstanding of token design expertise.
2. Due to the need for decentralization and consensus, each policy change is a very costly communication process, so it needs to be prudent and professionally audited, or have sufficient successful practice references, so that the community can fully understand and confident in its direction and correctness. Thus, more details objective review of the proposal items need to be conduct carefully for the purpose of public good and competitive advantage over other networks, not just subjective feelings and interests.
3. Effective public opinion research and governance mechanisms to achieve technically upgradeable means.

What needs to be added is that sufficient consensus does not mean that everyone is satisfied. Once the narrative of progress and change is adopted by Darwinia, rather than the unchanging narrative of Bitcoin, any change in the token economy means balance and trade-offs. The dispute of professional ideas may be resolved through communication and learning, but the conflict of interest conflicts can only be resolved through compromise, gradual progress or abandonment, and may not be resolved fundamentally, for example, ASIC miner in Ethereum will never agree the change of POW to POS, but Ethereum give a gradual period/hardforks to let them have time to reduce hardware invests.

**Reply by SasoLithops** [reply]
You mean agree to disagree😉 Absolutely true. Due to this changes some will stay some will leave that is unavoidable. 
I just wish we do proper thinking and invest many ideas into what would benefit Darwinia most. And then vote for it. I am not that smart to do all the calculations,but I will try and help with research as much as I can and provide feedback. I have been looking at recent changes of tokenomics in many projects. I will DEFINITELY not copy paste ideas as Darwinia has its own agenda. But I will try to post this weekend more detailed thoughts on what I belive could be a positive change. 

Hope many community members express their opinions on ideas proposed. I will agree with final decision that gets most votes if more thought is put into this. I wish only best for Darwinia.

**Comment by SasoLithops** [comment]
Also to add I would suggest rewarding voters as having active governence is very important on proposals. I also suggest we add a quorum that must be reached for proposal to pass. For start we could have 5% as I think only few vote. Ammount of Kton could be used for voting and rewards could be in Kton as well(the one we get for staking Ring from community pool). Let's start giving Kton more utility. I haven't been active there yet too...
And definitely give greater rewards to liquidity providers on DEXs we plan to list more then stakers as their risk is much greater.
I would go as far as stakers get 10%(we already get rewarded when locking Ring,with receiving Kton) as they do not sevure the chain any more. Voters could get 5% Ring as bonus next to Kton. The rest 75% go to community pool which rewards liquidity providers with 15% value. Do we have an address to our community pool so we can check the value? I would suggest we think of hackathlon soon so we unboard some defi projects or add additional support to Snowswap team which is doing great.

**Reply by hackfisher** [reply]
>  I also suggest we add a quorum that must be reached for proposal to pass. 
Quorum is definitely an important parameter, it is related to the turnout rate which relay on the community effective. On the other hand, Governance mechanisms need to consider adequate participation rates and effectiveness.

If we pursue the voting participation rate out of one-sidedness, it may lead to loss of efficiency and cause governance interruption or failure, so that the network itself loses the ability to revolute as a whole, which in turn affects competitiveness. Therefore, we need to pay attention to these two conveniences at the same time, and make the quorum as reasonable as possible. At the same time, we need to set up some professional collectives to improve efficiency. For example, the council can help to fast propose, while the technical commitee can fast track or veto, neither of them should hard power on governance. Another good practice to address this is the recently adoption of Polkadot's OpenGov, to use a Curve to describe the approval and quorum in the referenda for treasury spend at the same time.

> For start we could have 5% as I think only few vote. Ammount of Kton could be used for voting and rewards could be in Kton as well(the one we get for staking Ring from community pool). Let's start giving Kton more utility. I haven't been active there yet too...

I think keep the design and protocol simple is important, having two tokens in network governance at the same time will lead to more internal weight disputes and internal friction,, and KTON is derivative from RING from the perspective of long-term commitment to promote, but tokenized and transferrable, sole KTON holder may not have the conviction to RING if the KTON is bought from someone else. But we know when you voting in the governance pallets, you must conduct some conviction by locking the tokens for a duration, longer duration means multiply voting balance.

Alternatively, I think it is a good idea and better way for KTON holders to establish a DAO as the members of the govenance, just like DCDAO, sponsored by the nertwork treasury for their extra contributing to the public good of the network.

> And definitely give greater rewards to liquidity providers on DEXs we plan to list more then stakers as their risk is much greater.
> I would go as far as stakers get 10%(we already get rewarded when locking Ring,with receiving Kton) as they do not sevure the chain any more. Voters could get 5% Ring as bonus next to Kton. The rest 75% go to community pool which rewards liquidity providers with 15% value. Do we have an address to our community pool so we can check the value?

DEXs like snowswap are community dApps, for the purpose of ecosystem growth, there might be proposals or distrubution to support them from period to period, but this should not be a long term sustainable case.

For the incentive of voting, I have some doubts about in it, although I can't give a rationale opinion at present, others are welcome to comment.

**Reply by SasoLithops** [reply]
Maybe we could have a few not so important proposals soon to see the participation. I am active in one other DAO and quorum 30% is hardly reached. It failed two times due to whales not interested in governance at all. And those are much more active communities then ours. 

I still think Kton would be better then Ring as it is commitment token after all. And why have dual token system if there is not much value to Kton other then staking power. Locked Kton could be used for governance like it was in Evolution Land. If one wants to get more governance power  he still needs to buy Kton which due to poor liquidity is not easy to achive. And investing in Kton to have more power I think means caring for the project. One of our more active community members has almost 2000 Kton and he cares about Darwinia. 
Still no one can guarantee if whale with many Kton or many Ring will care. So nit sure why Ring is better, but am very much open for discussion.

On the other hand I really like your idea of KTONDAO😀

I agree not having long term support for community driven projects,many just care about "quick buck" but I fell like time to reward the stakers only has passed,since stakers do nothing now. They just collect Ring. Sad story is many will argue they have Ring locked for years and they are absolutely right. 
However for example I get double staking rewards then I did in Darwinia 1.0 for doing nothing(I no longer help in securing network) while Dustin on the other hand is the only one with decent ammount of Ring and Kton  providing liquidity in Snowswap DEX which many use and taking constant hit with impermanent loss. He should be rewarded way more then we are.

**Comment by SasoLithops** [comment]
So I have done some more thinking. This is my opinion where I see the problem and the solution.

This was my post on TG when we discussed how much should go to stakers and how much to treasury.

"" I think if team provided some decent calculations on why they chose the ratio,and what would the actual drop be,as so many stakers that do nothing the numbers would be much better.Now it feels as these numbers are chosen randomly.
I would much rather have healthy and rich treasury and Ring priced 0,1$ then small treasury and Ring 0,001$.
Why receive 1000 Ring at 0,001$ and dropping when I can have 100 Ring at 0,1$ and rising.
There are so many variables in crypto,that our own interest should not blind us...it is only one variable....I look forward to team discussion on Github....""

I am sad to see it went exactlly like this. I am receiving twice the rewards(so 20% should actually be ok) and Ring price keeps tanking.

Decent math calculations based on many variables should be made. One variable math does not work here and calculations were proven wrong.
Our APY for staking is not sustainsble on the long run and it is another factor driving price down.

Let me explain.
I think our inflation is atm not too high only due to low prices(that goes for Crab especialy). Othervise it is too high.
I feel that we have newly minted Ring and on the other side poor usecases for Ring.

So Ring is minted from staking. What can we use it for:

- Staking
- Fees

What else? We have no deep liquidity pools on DEXs(hopefully incentivesed from both sides), it is not traded on many exchanges, no Defi with Ring, no NFT market....So I see the issue here. We need to make Ring more usable/functional. People are not willing to lock tokens for few years any more...crypto evolves too quickly...most don't want to lock it for Crowdloans and that is 2 years. They want massive returns in Crowdloan rewards to give it up for 2 years, which is hardly feasible nowadays....

Our APY for staking I also think is not sustainsble on the long run and it is another factor driving price down.

So I would consider looking into making us use Ring for other purposes where we can see it being taken out of market and put into different uses.

**Reply by Laki389** [reply]
I have replied at https://github.com/orgs/dcdao/discussions/14 because it doesn't make sense  in writing the latest reply here while 6-7 comments and replies are missing.

**Reply by SasoLithops** [reply]
Only yours and one more mine are missing. And from Maduksjay. Copy it here,we will discuss it here as it is more about tokenomics now.

**Reply by SasoLithops** [reply]
- Annual inflation of 2%. Even better, every year a fixed amount of the current 2%, so every year a new 45 679 075 Ring. The first year it is 2%, the next year less than 2% and so on.
- Now, this is not enough and the only solution. Some project can be deflationary and the price of the coin can sink. In order for a project to be successful, it should have a product that is needed on the market and in demand. Does Darwinia have a product? I don't think there is yet.
- After the project has a product that is needed on the market, marketing is also needed. This means attending the most important gatherings like Polkadot Decoded, giving interviews for The Kusamarian (if there are funds and with significant crypto-global "influencers"), an aggressive presence on Twitter and so on.

I posted Laki reply above here. Will discuss more here.

As for first point:

IMPORTANT This calculations will be one dimensional as they consider constant price of 0.0025$ and constant number of collators staying at 16.IRL those two variables are subject to change. I will calculate in $ ammount as collators cost is in dollars.

So annual inflation of 2% gives as to your calulated Ring ammount per year cca 114200$ ammount.

44680$ ammount would go to us and collators.

Now collators cost is 60$ per month. So only their cost is cca 11520$. They must earn some too, so I think they deserve at least same ammout so that goes to 23040$.

That leaves 21640$ for nominators. Which is 60$ per day. And to make it back in Ring 23700 Ring per day. Now to callculate with total power it gives 34,7 RING per 1 million total stake power per day. 

Here it might go to 50 if some collators can go with little profit, which I doubt....

Are you ok with those rewards? 
And three most important variables not taken into account( price, total staked power, number of collators )

To second point:

My personal opinion is I don't like deflationary tokenomics, more with usecases of tokens( staking,Defi, fees, Nfts...)
Darwinia has one very succsessful project Subscan.

**Reply by Laki389** [reply]
> As for first point:
> 
> IMPORTANT This calculations will be one dimensional as they consider constant price of 0.0025$ and constant number of collators staying at 16.IRL those two variables are subject to change. I will calculate in $ ammount as collators cost is in dollars.



- I counted a total of 15 collators on Darwinia 2.



> 
> So annual inflation of 2% gives as to your calulated Ring ammount per year cca 114200$ ammount.
> 
> 44680$ ammount would go to us and collators.

It is 45 680$, but yeah, correct.
If 80% goes to Treasury, they will be 22 840$.


> 
> Now collators cost is 60$ per month. So only their cost is cca 11520$. They must earn some too, so I think they deserve at least same ammout so that goes to 23040$.
> 
> That leaves 21640$ for nominators. Which is 60$ per day. And to make it back in Ring 23700 Ring per day. Now to callculate with total power it gives 34,7 RING per 1 million total stake power per day.
> 
> Here it might go to 50 if some collators can go with little profit, which I doubt....
> 
> Are you ok with those rewards? And three most important variables not taken into account( price, total staked power, number of collators )

- Of course, these are wonderful rewards.



> To second point:
> 
> My personal opinion is I don't like deflationary tokenomics, more with usecases of tokens( staking,Defi, fees, Nfts...) Darwinia has one very succsessful project Subscan.

- Subscan is a project of Itering. Darwinia does not yet have any finished product.

**Comment by SasoLithops** [comment]
Been doing some more researching regarding our inflation.

Astar gave great idea about dynamic inflation (below at the thread)

https://forum.astar.network/t/astar-tokenomics-2-0-a-dynamically-adjusted-inflation/4924

Astar has its dApps staking so different model, but maybe we could think in that direction too, to have it price orientated.

I think Cosmos Hub has it too,but its inflation is too high to take into consideration...

Let me know if we could go down that path,will do more research...

Imho Megan would be a great person to discuss this with too....

**Comment by danairey** [comment]
Hi , great conversation you guys are having. My thoughts are you remove the staking yields or reduce them dramatically you will get the biggest wall of selling. I don't know the stats on how many people sell their yield or if they re-lock them , that would be the most important stat. I personally just re lock so I'm never selling, but if you reduced the yields I personally would then be a seller permanently. 

Only one solution and that's for Darwinia to have a product, chain people actually use , and then burn the transaction fees. Hopefully the chain is used so much, the burn outweighs inflation.  Right now Ring price is irrelevant , bar the treasury funding. Everything needs to be about USE CASE and burns; then token automically flies. 

This world we lived in where tokens could go up no matter what is done, and crypto will evolve into tokens that actually carry real value; that can derive value from actually being useful and that use awarded to token holders in the form of burns. Dot has a massive problem in terms of no one needs the DOT token, the system they have designed is totally flawed in terms of its own token. Ethereum have thought and created real use, along with a burn and hence people love it.

**Reply by SasoLithops** [reply]
Hi Dan!

Good points. Exactlly agree with you why the Dot doesn't moon(yet) compared to ETH which is used everywhere.
However I think it would be hard for Darwinia to achive Ethereum inflation/burn ratio as we most probably won't have usecase(s) that would generate that much fees as our inflation is simply to high atm regarding tokens(not $ value) that burning mechanism can/could cover. We could raise fees by a lot(which could now regarding $ value actually be done) but this would be bad decision imho as it would punish the users using for example Snowswap DEX, Helix...

One question. Not sure if that would be scenario, but would you still be selling if inflation was dropped towards  5-10% range BUT the price of token would start rising due to less RING being available in market?
I just think cca 50% apy is simply unsustainable on long term and community(us) will sooner or later need to accept that. Those days are over😥

To be honest I have been swapping Ring to Dot for past few months to not lose my $ value(but I plan to swap it back once we find good solution), but I don't have large ammounts so I don't affect the price if I would I would I would NOT do that...Hope whales don't do it,but I doubt it...
I would also like to see data as you wrote how much staked rewards(yield) are sold/swapped. And maybe data what causes larger price drops....

I think massive sell out in staking rewards is happening now,due to current model. But I absolutely could be wrong before seeing that data...I base all of this in constant price drop.

**Comment by danairey** [comment]
All valid points. Inflation could definitely be reduced, don't think it would be a major issue. Even though I don't see the token going up. Token only goes up once darwinia have a product people actually use. The guys involved with Darwinia are clearly real developers and that's always what I have loved about this project, its not been about the talk, or pushing price, its been about developing actual product. Just need people to use it. The fee discussion is always an interesting one, and it can be argued from both sides. I personally think at this stage higher fees is better, because I think the burn is more important. Eth is the most expensive chain on the block and its only that expensiveness that actually creates its value. But then the chain is also running close to max at times and therefore can be expensive. Not saying Darwinia should be expensive but fees have room to be increased if they burnt. If a blockchain can really scale, then fees could be reduced but no one has been able to achieve that so far. Suppose you have to work out what the worth of your chain is in terms of fees and get it to the max people are willing to pay.

**Comment by SasoLithops** [comment]
I agree. Imo at current price levels  I think reduced inflation wouldn't make a difference too.
It was exactlly why I liked Darwinia too. All about the tech. But maybe that is us who found Darwinia in 2019/2020. Not sure new people that learn about it will look at it like we did back then. Crypto has evolved so much since then when we loved the project if it had good idea for future and had nice tokenomics. Now like you said people look for projects with real value. That should be the start point imho.
Definitely plenty of room in fees charge👍Good example to learn from burning fees is also Vechain VTHO. It really needs huge ammount of transactions for burning to make sense. They strugling too...

**Comment by wuminzhe** [comment]
I feel that it is intuitive to burn the transaction fee like ethereum.

<img width="371" alt="image" src="https://github.com/darwinia-network/darwinia/assets/1608576/8bd3c4b7-a061-43b8-9fcd-b7cd968ac790">

**Reply by SasoLithops** [reply]
I agree. However I feel like with our current transactions fees the ammount of Ring burned will not be high enough to be comparable to ETH. Do we have the data how much Ring is used for transactions (day/month/year)? I can't find it.

**Reply by wuminzhe** [reply]
no such data now. but let me see if it is hard to do.

**Reply by SasoLithops** [reply]
Awesome. Imho this data would help us in future analysis.

**Reply by Laki389** [reply]
> no such data now. but let me see if it is hard to do.

In Darwinia 2 (past 136 days) there are a total of 6 709 transfers and 2 406 signed extrinsics (source: Subscan).

That's an insignificant amount of $ring.

**Reply by SasoLithops** [reply]
Good job! There are some transactions on other chains as well,but they don't come close to DSC. 
So if we think of burning tokens from fees at this time imo is not the way. It could be done,but little effect. 
I see how VTHO is strugling for years with lack of usage. Usecases need to be huge like on Ethereum and fees high for that to make sense. But high fees is definitely a turn off. I stopped using ETH because of that fact.5-10$ fees on weekends are a lot for many of us users....

Maybe we could introduce something that burns Ring. Like for example you could either lock Ring to earn KTON or burn some ammount of Ring next to locking it to lower your locked Ring period.

**Comment by hackfisher** [comment]
Make a periodic summary, this RFC basically includes two points. First is to draw lessons from EIP1559 by introducing a burning mechanism, and second is to indirectly reduce the inflation rate by adjusting the annual new issuance. Currently, there are still disagreements in the community about the second point, which requires further discussion and review. Regarding the first point, it has already been adopted and accepted in the latest Genepaper V4 and DIP-3.

Although DIP-3 and EIP1559 have minimal impact on the current economic data, they represent a significant change in the token economic design philosophy. This will affect narrative expression and expectations. By providing more data analysis and motivation through the community, it can promote and guide the community to allocate resources and invest in areas that can help burn more tokens or contribute to revenue, achieving a return to long-term token economic balance.

Ref:

1. https://darwinia.network/Genepaper_v4.pdf
2. https://github.com/darwinia-network/DIPs/blob/main/DIPs/dip-3.md
